src/pages/groupComparison/GroupComparisonMutationTable.tsx

```diff
@@ -92,6 +84,54 @@ export default class GroupComparisonMutationTable extends MutationTable<
         super(props);
     }
 
+    // determine what tumor type attribute is the same for data
+    protected getSameTumorTypeAttribute(
+        data: Mutation[]
+    ): TumorTypeAttribute | undefined {
+        // first, return CANCER_TYPE_DETAILED if the same in all mutations
+        if (
+            _.every(
+                data,
+                m => this.resolveTumorType(m) === this.resolveTumorType(data[0])
+            )
+        ) {
+            return 'CANCER_TYPE_DETAILED';
+        }
+        // if not, return CANCER_TYPE if the same in all mutations
+        if (
+            _.every(
+                data,
+                m =>
+                    this.resolveCancerType(m) ===
+                    this.resolveCancerType(data[0])
+            )
+        ) {
+            return 'CANCER_TYPE';
+        }
+    }
+
+    // determine what function to use to resolve tumor type based on what tumor type attribute is the same for data
+    protected getResolveTumorType(data: Mutation[]) {
+        if (this.getSameTumorTypeAttribute(data) === 'CANCER_TYPE_DETAILED') {
+            return this.resolveTumorType;
+        }
+        if (this.getSameTumorTypeAttribute(data) === 'CANCER_TYPE') {
+            return this.resolveCancerType;
+        }
```

Comment:
this `resolveTumorType` vs `resolveCancerType` naming feels a little weird, tho i guess it was already there. Not super clear what the difference between those is

Reply:
yea, I wasn't sure what to name it but `resolveTumorType` typically gets the `Cancer Type Detailed` from a mutation while the new `resolveCancerType` will retrieve the `Cancer Type`. any ideas on name changes? throughout the codebase, the term "Tumor Type" is used but mostly just for "Cancer Type Detailed" and I didn't want to rename all of that.

Reply:
I see - yeah as long as it is consistent that's fine by me. If we were to change it throughout the code I'd prolly be explicit and just say resolveCancerType vs resolveCancerTypeDetailed